threedi_modelchecker/checks/raster.py

```diff
@@ -158,6 +160,32 @@ def description(self):
         return f"The file in {self.column_name} does not use a projected CRS."
 
 
+class RasterHasMatchingEPSGCheck(BaseRasterCheck):
+    """Check whether a raster's EPSG code matches the EPSG code in the global settings for the SQLite.
+
+    Only works locally.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get_invalid(self, session):
+        epsg_code_query = session.query(models.GlobalSetting.epsg_code).first()
+        if epsg_code_query is not None:
+            self.epsg_code = epsg_code_query[0]
+        else:
+            self.epsg_code = None
+        return super().get_invalid(session)
+
+    def is_valid(self, path: str, interface_cls: Type[RasterInterface]):
+        with interface_cls(path) as raster:
+            raster_epsg_code = raster.epsg_code
```

Comment:
We willen deze error niet zien als er:

- geen valid geotiff is
- geen projectie is de GeoTIFF staat

Daarom moet je hier zoiets doen:

```
if not raster.is_valid_geotiff or raster.is_geographic is None:
    return True
```

Het geval `self.epsg_code is None` moet je wellicht ook expliciet afdekken met:

```
if self.epsg_code is None:
    return False
```

Het geval `raster.epsg_code is None` is ook mogelijk, dan zou je ook `False` moeten teruggeven.

Reply:
Moet False ook worden teruggegeven als de raster epsg en global settings epsg allebei None zijn? Per slot van rekening zijn ze dan wel hetzelfde.